packages/sanes-browser-extension/src/index.tsx

```diff
@@ -33,20 +34,19 @@ const render = (
   );
 };
 
-getPersonaData().then(persona => {
-  const requests = getQueuedRequests();
-  render(Route, persona, requests);
+type PersonaContexType = [GetPersonaResponse, boolean];
 
-  hasStoredPersona().then(hasPersona => {
-    const hasRequests = requests.length > 0;
-    const url = initialUrl(!!persona, hasPersona, hasRequests);
-    history.push(url);
+Promise.all([getPersonaData(), hasStoredPersona()]).then(([persona, hasPersona]: PersonaContexType) => {
+  const requests = getQueuedRequests();
+  const hasRequests = requests.length > 0;
+  const url = initialUrl(!!persona, hasPersona, hasRequests);
```

Comment:
I think the variable name `hasPersona` is not optimal because `persona` can be `null` as well, so it is hard to understand here the difference is. Can you do the following renamings to make clear this is about the database?

1. `hasStoredPersona()` -> `getHasStoredPersona()`
2. `hasPersona` -> `hasStoredPersona`